### PR TITLE
Add a Go SDK

### LINKED ISSUE #70: Implement Go provider
We will add Go support to the Kubernetes provider.  This includes the following work:
- [x] Core Kubernetes API resources (https://github.com/pulumi/pulumi-kubernetes/pull/1029)
- [x] Helm/YAML support (https://github.com/pulumi/pulumi-kubernetes/issues/1037)
- [x] apiextensions.CustomResource (https://github.com/pulumi/pulumi-kubernetes/pull/1125)

### PR DESCRIPTION
**Pull request by lukehoban**
This is 99% the work of @lblackstone  and @pgavlin 🙏 .  

It is still WIP, but opening the PR as it is now in a usable state and ready for more eyes.

```go
package main

import (
	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/go/kubernetes/core/v1"
	"github.com/pulumi/pulumi/sdk/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := corev1.NewPod(ctx, "foo", &corev1.PodArgs{
			Spec: corev1.PodSpecArgs{
				Containers: corev1.ContainerArray{
					corev1.ContainerArgs{
						Name:  pulumi.String("nginx"),
						Image: pulumi.String("nginx"),
					},
				},
			},
		})
		if err != nil {
			return err
		}

		return nil
	})
}
```

Fixes #70.

### CODE REVIEW

File: cmd/pulumi-gen-kubernetes/main.go [hunk 1]
```diff
@@ -307,3 +311,32 @@ func writeDotnetClient(data map[string]interface{}, outdir, templateDir string)
...
 }
+
+func writeGoClient(data map[string]interface{}, outdir string) {
+	pkg := writePulumiSchema(data)
+	files, err := gogen.GeneratePackage("pulumigen", pkg)
+	if err != nil {
+		panic(err)
+	}
+	for filename, contents := range files {
+		//filename = strings.TrimPrefix(filename, "kubernetes/")
```
Comment (lukehoban) [inline cmd/pulumi-gen-kubernetes/main.go]: Remove commented code.

File: pkg/provider/provider.go [hunk 2]
```diff
@@ -170,6 +170,10 @@ func (k *kubeProvider) invalidateResources() {
 	k.resources = nil
 }
 
+func (k *kubeProvider) GetSchema(ctx context.Context, req *pulumirpc.GetSchemaRequest) (*pulumirpc.GetSchemaResponse, error) {
+	panic("implement me")
```
Comment (lukehoban) [inline pkg/provider/provider.go]: Is panicing "right" here - or can this just return `Unimplemented` explicitly?

File: pkg/gen/typegen.go [hunk 3]
```diff
@@ -110,16 +116,16 @@ func (vc VersionConfig) TopLevelKindsAndAliases() []KindConfig {
...
+		//if strings.HasPrefix(kind.APIVersion(), apiRegistration) {
+		//	alias := KindConfig{}
+		//	err := copier.Copy(&alias, kind)
+		//	if err != nil {
+		//		panic(err)
+		//	}
+		//	defaultAPIVersion := "apiregistration" + strings.TrimPrefix(kind.APIVersion(), apiRegistration)
+		//	alias.defaultAPIVersion = defaultAPIVersion
+		//	kindsAndAliases = append(kindsAndAliases, alias)
+		//}
```
Comment (lukehoban) [inline pkg/gen/typegen.go]: Remove commented code?

File: pkg/gen/typegen.go [hunk 4]
```diff
@@ -63,6 +61,14 @@ type GroupConfig struct {
 // Group returns the name of the group (e.g., `core` for core, etc.)
 func (gc GroupConfig) Group() string { return gc.group }
 
+// URNGroup returns a group version that can be used in a URN.
+func (gc *GroupConfig) URNGroup() string {
+	//if strings.HasPrefix(gc.group, apiRegistration) {
+	//	return "apiregistration" + strings.TrimPrefix(gc.group, apiRegistration)
+	//}
```
Comment (lukehoban) [inline pkg/gen/typegen.go]: Remove commented code?

**Comment by lukehoban** [comment]
Looks like I'm not allowed to "Approve" since I opened this - but LGTM :tada:.  @lblackstone you may need to sign off yourself to unblock the review checks.

File: pkg/provider/provider.go (hunk 2 above)
Comment (lblackstone) [reply pkg/provider/provider.go]: Updated to return Unimplemented error

**Comment by lblackstone** [comment]
Ok, just need to merge #1021 and then resolve merge conflicts with this PR and we should be good to go.

**Pull request merged**